librz/core/cmd/cmd_type.c

```diff
@@ -118,8 +122,12 @@ static RzCmdStatus type_format_print_variable(RzCore *core, const char *type, co
 		return RZ_CMD_STATUS_ERROR;
 	}
 	ut64 addr = rz_core_analysis_var_addr(core, var);
-	// TODO: Convert to the API
-	rz_core_cmdf(core, "pf %s @ 0x%08" PFMT64x "\n", fmt, addr);
+	ut64 oldoff = core->offset;
+	rz_core_seek(core, addr, true);
+	char *r = rz_core_print_format(core, fmt, RZ_PRINT_MUSTSEE);
```

Comment:
i think we should make `rz_core_print_format` accept an address and just load the data from there instead of having to do always seek/seek before/after. This is how I handled these kind of things in many other APIs.